pkg/controller/deployment/deployment_controller.go

```diff
@@ -84,8 +92,103 @@ func (d *DeploymentController) reconcileDeployment(deployment *extensions.Deploy
 }
 
 func (d *DeploymentController) reconcileRecreateDeployment(deployment extensions.Deployment) error {
-	// TODO: implement me.
-	return nil
+	oldRCs, err := d.getOldRCs(deployment)
+	if err != nil {
+		return err
+	}
+
+	// scale all old rcs to zero
+	waitScale := sync.WaitGroup{}
+	waitScale.Add(len(oldRCs))
+	errChan := make(chan error, len(oldRCs))
+	for _, oldRC := range oldRCs {
+		go func(rc *api.ReplicationController) {
+			defer waitScale.Done()
+			if _, err := d.scaleRCAndRecordEvent(rc, 0, deployment); err != nil {
```

Comment:
This will try to scale rcs that are already scaled down. Ensure that rc.spec.replicas is not zero before trying to scale down to zero.